modules/xphoto/test/test_inpainting.cpp

```diff
@@ -0,0 +1,75 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+using namespace xphoto;
+
+
+static void test_inpainting(const Size inputSize, InpaintTypes mode, double expected_psnr, ImreadModes inputMode = IMREAD_COLOR)
+{
+    string original_path = cvtest::findDataFile("cv/shared/lena.png");
+    string mask_path = cvtest::findDataFile("cv/inpaint/mask.png");
+
+    Mat original_ = imread(original_path, inputMode);
+    ASSERT_FALSE(original_.empty()) << "Could not load input image " << original_path;
+
+    Mat mask_ = imread(mask_path, IMREAD_GRAYSCALE);
+    ASSERT_FALSE(mask_.empty()) << "Could not load error mask " << mask_path;
+
+    Mat original, mask;
+    resize(original_, original, inputSize, 0.0, 0.0, cv::INTER_AREA);
+    resize(mask_, mask, inputSize, 0.0, 0.0, cv::INTER_NEAREST);
+
+    Mat mask_valid = (mask == 0);
+    Mat im_distorted(inputSize, original.type(), Scalar::all(0));
+    original.copyTo(im_distorted, mask_valid);
+
+    Mat reconstructed;
+    cv::xphoto::inpaint(im_distorted, mask_valid * (1/255.0), reconstructed, mode);
```

Comment:
> mask_valid * (1/255.0)

Looks like masks `{0, 1}` are accepted only. `{0, 255}` doesn't work good.

Please add mask pre-processing code into algorithm implementation, like documentation says:

> @param mask mask (CV_8UC1), where non-zero pixels indicate valid image area, while zero pixels
    indicate area to be inpainted	    indicate area to be inpainted

Try this pre-processing code:

```
threshold(mask_as_inputarray, mask_01, 0.0, 1.0, THRESH_BINARY)l
```

(after that please remove this fix from test code)